components/brave_wallet/browser/bitcoin_keyring.cc

```diff
@@ -0,0 +1,84 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/bitcoin_keyring.h"
+
+#include <memory>
+#include <utility>
+
+namespace brave_wallet {
+
+std::string BitcoinKeyring::GetReceivingAddress(uint32_t account_index,
+                                                uint32_t receiving_index) {
+  auto key = DeriveReceivingKey(account_index, receiving_index);
+  if (!key) {
+    return "";
+  }
+
+  HDKey* hd_key = static_cast<HDKey*>(key.get());
+
+  return hd_key->GetSegwitAddress();
+}
+
+std::string BitcoinKeyring::GetChangeAddress(uint32_t account_index,
+                                             uint32_t change_index) {
+  auto key = DeriveChangeKey(account_index, change_index);
+  if (!key) {
+    return "";
+  }
+
+  HDKey* hd_key = static_cast<HDKey*>(key.get());
+
+  return hd_key->GetSegwitAddress();
+}
+
+std::string BitcoinKeyring::GetAddressInternal(HDKeyBase* hd_key_base) const {
+  if (!hd_key_base) {
+    return std::string();
+  }
+  HDKey* hd_key = static_cast<HDKey*>(hd_key_base);
+  return hd_key->GetSegwitAddress();
+}
+
+std::unique_ptr<HDKeyBase> BitcoinKeyring::DeriveAccount(uint32_t index) const {
+  // m/84'/0'/{index}'
+  return root_->DeriveHardenedChild(index);
+}
+
+std::unique_ptr<HDKeyBase> BitcoinKeyring::DeriveReceivingKey(
```

Comment:
nit: This function looks very similar to the one below. Would it make sense to refactor them together so that they call an internal function called DeriveKey that takes in a parameter for the child index that's passed to DeriveNormalChild?

Reply:
It makes sense to refactor new code when whole feature gets in production and receives some updates. Right now I'm not sure where it all we go. Maybe similarity goes away and refactoring should be reverted hence double work.
I'm personally for some copypasting unless there is 110% confidence that similar code needs merge.
I'll add a TODO.